### PR TITLE
Calculator

### LINKED ISSUE #1: Calculator
Implement:
- Basic operations
- Clear
- +/- button
- Backspace

### PR DESCRIPTION
**Pull request by viniciusbenite**
Calculator implemented:

    [ ✓] - Math operations (+, -, *, /);
    [ ✓] - Sign change (+/-);
    [ ✓] - Clear and backspace;
    [ ✓] - Basic memory;

### CODE REVIEW

File: calculator/calculator.cpp [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    , ui(new Ui::Calculator)
+{
+    // Create the UI (Display)
+    ui->setupUi(this);
+
+    // The default value of display: 0.0
+    ui->display->setText(QString::number(calcValue));
+
+    QPushButton *numButtons[10]; // Numbers 0 to 9
+
```
Comment (andre-exd) [inline calculator/calculator.cpp]: Please remove this extra line. That way it is more noticeable that numButtons are being initialized in the for loop

File: calculator/calculator.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    {
+        ui->display->clear();
+        displayFlag = false;
+    }
+
+    QPushButton *btn = (QPushButton *)sender();
+    QString btnValue = btn->text();
+    QString displayValue = ui->display->text();
+
+    if ((displayValue.toDouble() == 0) || (displayValue.toDouble() == 0.0))
```
Comment (andre-exd) [inline calculator/calculator.cpp]: Are both verifications really necessary? I would assume default value of double should always be 0.0. Can you confirm?

File: calculator/calculator.cpp [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    QPushButton *btn = (QPushButton *)sender();
+    QString btnValue = btn->text();
+    double doubleCurrentValue = currentvalue.toDouble();
+
+    if (divTrigger || multiTrigger || addTrigger || subTrigger)
+    {
+        if (divTrigger)
+        {
+            result = calcValue / doubleCurrentValue;
+        } else if (multiTrigger)
```
Comment (andre-exd) [inline calculator/calculator.cpp]: Can you fix the indentation of the else ifs in EqualButton()?

File: calculator/calculator.cpp [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    displayFlag = true;
+}
+
+void Calculator::ChangeNumberSign()
+{
+    QString currentDisplay = ui->display->text();
+
+    QRegExp reg("[-]?[0-9.]*");
+
+    if (reg.exactMatch(currentDisplay))
```
Comment (andre-exd) [inline calculator/calculator.cpp]: What is the purpose of this verification? In which cases don't you want to change the number sign?

File: calculator/calculator.cpp [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+        if (divTrigger)
+        {
+            result = calcValue / doubleCurrentValue;
+        } else if (multiTrigger)
+        {
+            result = doubleCurrentValue * calcValue;
+        } else if (addTrigger)
+        {
+            result = doubleCurrentValue + calcValue;
+        } else if (subTrigger)
```
Comment (andre-exd) [inline calculator/calculator.cpp]: This else if is not consistent with last else on MathButtonPressed()

**Review by andre-exd — CHANGES_REQUESTED** [review]

File: calculator/calculator.cpp (hunk 3 above)
Comment (viniciusbenite) [reply calculator/calculator.cpp]: Done.

File: calculator/calculator.cpp (hunk 1 above)
Comment (viniciusbenite) [reply calculator/calculator.cpp]: Done.

File: calculator/calculator.cpp (hunk 2 above)
Comment (viniciusbenite) [reply calculator/calculator.cpp]: No. toDouble() take cares of it.

File: calculator/calculator.cpp (hunk 5 above)
Comment (viniciusbenite) [reply calculator/calculator.cpp]: I didn't understand. The identation?

File: calculator/calculator.cpp (hunk 4 above)
Comment (viniciusbenite) [reply calculator/calculator.cpp]: You can type anything  you want in the display. I have to move this verification up.

**Review by andre-exd — APPROVED** [review]

**Pull request merged**